sklearn/neighbors/binary_tree.pxi

```diff
@@ -1054,6 +1071,16 @@ cdef class BinaryTree:
         self.data_arr = np.asarray(data, dtype=DTYPE, order='C')
         self.data = get_memview_DTYPE_2D(self.data_arr)
 
+        self.new_data_arr = check_array(data, order='C', dtype=[np.float64])#, np.float32])
```

Comment:
Why do we need this as well as `data_arr`? And why is the option of float32 commented out?